src/battle/star/peach_beam_7997B0.c

```diff
@@ -1,26 +1,40 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "battle/star/peach_beam_7997B0", func_802A1000_7997B0);
-
-INCLUDE_ASM(s32, "battle/star/peach_beam_7997B0", func_802A10AC_79985C);
-
-INCLUDE_ASM(s32, "battle/star/peach_beam_7997B0", func_802A116C_79991C);
+#define NAMESPACE battle_star_peach_beam
 
-INCLUDE_ASM(s32, "battle/star/peach_beam_7997B0", func_802A1218_7999C8);
+#include "common/UnkStarFuncs.inc.c"
 
-#define NAMESPACE battle_star_peach_beam
 #include "common/FadeBackgroundToBlack.inc.c"
 
-INCLUDE_ASM(s32, "battle/star/peach_beam_7997B0", func_802A137C_799B2C);
+#include "common/UnkBackgroundFunc2.inc.c"
 
-INCLUDE_ASM(s32, "battle/star/peach_beam_7997B0", func_802A1414_799BC4);
+#include "common/UnkBackgroundFunc.inc.c"
 
 INCLUDE_ASM(s32, "battle/star/peach_beam_7997B0", func_802A1494_799C44);
 
-INCLUDE_ASM(s32, "battle/star/peach_beam_7997B0", func_802A14E8_799C98);
+#include "common/SetNpcCollision32.inc.c"
 
 INCLUDE_ASM(s32, "battle/star/peach_beam_7997B0", func_802A1518_799CC8);
 
-INCLUDE_ASM(s32, "battle/star/peach_beam_7997B0", func_802A2468_79AC18);
+ApiStatus func_802A2468_79AC18(ScriptInstance* script, s32 isInitialCall) {
+    script->varTable[0] = D_802A3838;
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802A247C_79AC2C(ScriptInstance* script, s32 isInitialCall) {
+    if (isInitialCall) {
+        script->functionTemp[0].s = 230;
+    }
+
+    set_background_color_blend(0, 0, 0, script->functionTemp[0].s);
+
+    script->functionTemp[0].s -= 5;
+    if (script->functionTemp[0].s <= 200) {
+        set_background_color_blend(0, 0, 0, 200);
+        return ApiStatus_DONE2;
+    }
+
+    return ApiStatus_BLOCK;
+}
 
-INCLUDE_ASM(s32, "battle/star/peach_beam_7997B0", func_802A247C_79AC2C);
+// this file is identical in c code to star beam
```

Comment:
like, identical identical?
the scripts must be different, perhaps we should play about with includes then

Reply:
yeah